x-pack/plugins/apm/public/components/shared/span_links/span_links_table.tsx

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiBasicTableColumn,
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiCopy,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiInMemoryTable,
+  EuiLink,
+  EuiPopover,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import { SpanLinkDetails } from '../../../../common/span_links';
+import { asDuration } from '../../../../common/utils/formatters';
+import { useApmParams } from '../../../hooks/use_apm_params';
+import { useApmRouter } from '../../../hooks/use_apm_router';
+import { ServiceLink } from '../service_link';
+import { getSpanIcon } from '../span_icon/get_span_icon';
+
+interface Props {
+  items: SpanLinkDetails[];
+}
+
+export function SpanLinksTable({ items }: Props) {
+  const { link } = useApmRouter();
```

Comment:
I think it's better to use `router.link` directly, it's not a utility method but a method of a class instance. it might not cause an issue today but maybe at some point it will when the implementation of `router.link` references `this`.